File: ISS Emergency/Assets/Scripts/TriggerPanelManager.js
#pragma strict

// Hannes Helmholz
//
// 


@script RequireComponent(NetworkView); // for RPC

private class TriggerPanelManager extends MonoBehaviour {

	// =============================================================================
	// MEMBERS ---------------------------------------------------------------------
	private var NetView:NetworkView;
	private var State:boolean = false;
	private var Panels:TriggerPanel[];
	// ============================================================================
	
	
	
	// =============================================================================
	// METHODS UNITY ---------------------------------------------------------------
	
	function Awake() {
		NetView = networkView;
	
		Panels = gameObject.GetComponentsInChildren.<TriggerPanel>();
	}
	
	// =============================================================================
	
	
	
	// =============================================================================
	// METHODS ---------------------------------------------------------------------

	public function SetState(state:boolean) {
		NetView.RPC("RPCSetState", RPCMode.AllBuffered, state);
	}
	
	
	private function SetPanels(state:boolean) {
		for (var panel:TriggerPanel in Panels)
			panel.SetState(state);
	}
	// =============================================================================
	
	
	
	// =============================================================================
	// METHODS RPC -----------------------------------------------------------------
	
	@RPC
	function RPCSetState(state:boolean) {
		State = state;
		
		SetPanels(State);
	}
	// =============================================================================
}

File: Assets/Scripts/TriggerPanelManager.cs

// Hannes Helmholz
// Dieser TriggerManager managt alle Triggerpanels und stellt sicher das diese mit einander interagieren.
// 

using UnityEngine;

[RequireComponent (typeof(NetworkView))] // for RPC

public class TriggerPanelManager : MonoBehaviour
{

    // =============================================================================
    // MEMBERS ---------------------------------------------------------------------
    private NetworkView NetView;
    private bool State = false;
    private TriggerPanel[] Panels;
    // ============================================================================
 
 
 
    // =============================================================================
    // METHODS UNITY ---------------------------------------------------------------
 
    protected virtual void Awake ()
    {
        NetView = networkView;
 
        Panels = gameObject.GetComponentsInChildren<TriggerPanel> ();
    }
 
    // =============================================================================
 
 
 
    // =============================================================================
    // METHODS ---------------------------------------------------------------------

    public void SetState ( bool state )
    {
        NetView.RPC ( "RPCSetState", RPCMode.AllBuffered, state );
    }
 
 
    private void SetPanels ( bool state )
    {
        foreach ( TriggerPanel panel in Panels )
            panel.SetState ( state );
    }
    // =============================================================================
 
 
 
    // =============================================================================
    // METHODS RPC -----------------------------------------------------------------
 
    [RPC]
    void RPCSetState ( bool state )
    {
        State = state;
     
        SetPanels ( State );
    }
    // =============================================================================
}